src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -1816,6 +1816,9 @@ CHIP_ERROR ExtractVIDPIDFromX509Cert(const ByteSpan & certificate, AttestationCe
     }
 
 exit:
+    ASN1_OBJECT_free(commonNameObj);
```

Comment:
@andy31415 Why is #19819 blocked on this?  I thought we bumped the stack size to avoid this problem there?

I'm not comfortable with fast-tracking these sorts of changes.  Furthermore, I think the non-test changes here should be broken out into a separate PR so they don't get lost in the huge test diff.  I'd also really like to understand what made these non-test issues come to light in this PR....

Reply:
It spiraled:
   - 19189 was running out of stack because test contexts got larger, so Tennessee asked me to create a PR to move all contexts to heap
   - this exposed a ASAN failure (the CASESession accessing a freed ExchangeManager because TestCASESession is global and the contexts were moved to heap)
   - this got me running is_asan locally and that seems to leak detect too for me (maybe because not clang, not sure, but it looks like legitimate leaks)
   
 So now I have:
   - move contexts to heap
   - changes to fix memory leaks in tests and this is the only non-test one

Reply:
OpenSSL seems to leak 3 objects every time we call ExtractVIDPIDFromX509Cert. Seems bad.

Reply:
I removed the fast track - originally added it because it was "test only" changes. But tests found bugs too

Reply:
Regarding why these came out, I believe my asan settings were somehow stricter - CI found the exchange manager crash but not leaks, but locally I found leaks and fixed those too.

Reply:
> this exposed a ASAN failure (the CASESession accessing a freed ExchangeManager because TestCASESession is global and the contexts were moved to heap)

I'm not following this part.  TestCASESession is a function...  Is this about `CASEServerForTest`?

If so, then we should hang it as a member off the context (which should then inherit from `Test::LoopbackMessagingContext` but have this extra member), so our lifetimes can be sane.  Or make it a pointer and allocate it in `CASETestSecurePairingSetup` and destroy in `CASE_TestSecurePairing_Teardown`.  Because having tests do differrent lifetime ordering than our real code is not very useful...

> Regarding why these came out, I believe my asan settings were somehow stricter - CI found the exchange manager crash but not leaks

I wish we could figure out why and fix CI to catch those....  Are they possibly showing up in CI but not being treated as fatal or something?  Would be good to figure out what's going on.

In any case, I think we should have a PR to fix the bugs, and a separate PR for the test mass-change.

Reply:
`CASEServerForTest` is a `CASEServer` which will access mExchangeManager on ::Shutdown() which is called in the destructor.

So moving a lifetime of exchange manager to be less then CASEServer will make an illegal memory access. So existing code relies on static destructor order, which is also not ok but happens to work.

Reply:
Split out bug fixing into https://github.com/project-chip/connectedhomeip/pull/19953

Still need to keep this PR since it requires those bug fixes to pass.

Reply:
@andy31415 My point was that CASEServerForTest should have a _smaller_ lifetime than the exchange manager.  And I proposed a specific way to accomplish that above....